lua/cfc_prop_restoration/core/sv/sv_adinterface.lua

```diff
@@ -0,0 +1,136 @@
+ADInterface = {}
+
+local areacopy_classblacklist = {
+    gmod_anchor = true
+}
+
+local function PlayerCanDupeCPPI( ply, ent )
+    if ent.DoNotDuplicate or areacopy_classblacklist[ent:GetClass()] or not IsValid( ent:GetPhysicsObject() ) or not duplicator.IsAllowed( ent:GetClass() ) then return false end
+    return ent:CPPIGetOwner() == ply
+end
+
+local phys_constraint_system_types = {
+    Weld = true,
+    Rope = true,
+    Elastic = true,
+    Slider = true,
+    Axis = true,
+    AdvBallsocket = true,
+    NoCollide = true,
+    Motor = true,
+    Pulley = true,
+    Ballsocket = true,
+    Winch = true,
+    Hydraulic = true,
+    WireMotor = true,
+    WireHydraulic = true
```

Comment:
Typically with a tall table like this, it's a good idea to alphabetize them so that it's easy to find what you're looking for later

Reply:
I put this file together from sections of the adv dupe 2 code, I'll go through and refactor it all later today